### PR TITLE
Zap Protocol Balances

### PR DESCRIPTION
**Pull request by blurpesec**
### Description

Added in the DeFi protocol balances

## Todos:
- [x] ~Waiting for an api endpoint for the s3 item, then update it in `DeFiReserveMapService`~
- [x] ~Above api endpoint has cors issues on localhost. passed on to mia~ :dagger: . Resolved

### CODE REVIEW

File: common/v2/services/ApiService/DeFiReserveMap/DeFiReserveMap.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { AxiosInstance } from 'axios';
+import { default as ApiService } from '../ApiService';
+
+let instantiated: boolean = false;
+
+// TODO: Change this once api endpoint becomes avail
+const DEFI_RESERVE_MAPPING_URL = 'https://defi.mycryptoapi.com';
```
Comment (GuillaumeRx) [inline common/v2/services/ApiService/DeFiReserveMap/DeFiReserveMap.ts]: We should export this in the config

File: common/v2/services/ApiService/DeFiReserveMap/DeFiReserveMap.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      instantiated = true;
+    }
+  }
+
+  public getDeFiReserveMap = async () => {
+    try {
+      return this.service.get('').then(res => res.data);
+    } catch (e) {
+      throw e;
+    }
```
Comment (GuillaumeRx) [inline common/v2/services/ApiService/DeFiReserveMap/DeFiReserveMap.ts]: I imagine the `async` will be usefull when you will complete this

File: common/v2/services/RatesProvider.tsx [hunk 3]
```diff
@@ -87,6 +108,17 @@ export function RatesProvider({ children }: { children: React.ReactNode }) {
     })();
   });
 
+  useEffectOnce(() => {
+    (async () => {
+      const value = await mounted(
+        fetchDeFiReserveMappingList().then(e => {
+          return e;
```
Comment (GuillaumeRx) [inline common/v2/services/RatesProvider.tsx]: you can reduce that by doing `e => e`

File: common/v2/services/ApiService/DeFiReserveMap/DeFiReserveMap.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+export default class DeFiReserveMapService {
+  public static instance = new DeFiReserveMapService();
+
+  private service: AxiosInstance = ApiService.generateInstance({
+    baseURL: DEFI_RESERVE_MAPPING_URL
+  });
+
+  constructor() {
+    if (instantiated) {
+      throw new Error(`DeFiReserveMapService has already been instantiated.`);
```
Comment (GuillaumeRx) [inline common/v2/services/ApiService/DeFiReserveMap/DeFiReserveMap.ts]: Could we use translation for this ?

Comment (blurpesec) [reply common/v2/services/ApiService/DeFiReserveMap/DeFiReserveMap.ts]: yeh. ill add it in the translation pr

**Pull request merged**